onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java

```diff
@@ -824,14 +832,25 @@ void runSimulation(int expectedGetReadyIndex, int expectedPullCordIndex) {
             Boolean check1 = provider.getGetReady() && !provider.getFinished();
             assertTrue(check1);
 
+            if(expectedGetReadyIndex!= 0){
+                prevLocation = mLocations[expectedGetReadyIndex - 1];
+            }
+
             for (int i = expectedGetReadyIndex; i <= expectedPullCordIndex; i++) {
                 Location l = mLocations[i];
                 try {
                     Thread.sleep((mTimes[i] / SPEED_UP));
                 } catch (Exception e) {
                     Log.e(TAG, e.getMessage());
                 }
-                provider.locationUpdated(l);
+
+                // Code added to check for duplicate locations in the .csv log files
+                if (prevLocation == null) {
```

Comment:
Same as above - you can collapse to:

`if (prevLocation == null || !LocationUtils.isDuplicate(prevLocation, l)) {`